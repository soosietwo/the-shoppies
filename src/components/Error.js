import React from "react";
import { Banner } from "@shopify/polaris";

const Error = ({ title, details }) => (
  <Banner status="critical" title={title}>
    <p>{details}</p>
  </Banner>
);

export default Error;
