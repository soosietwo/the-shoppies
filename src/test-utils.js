// test-utils.js
import React from "react";
import { render } from "@testing-library/react";
import { AppProvider, Frame } from "@shopify/polaris";
import en from "@shopify/polaris/locales/en.json";
import Store from "./store";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllTheProviders = ({ children }) => {
  return (
    <AppProvider i18n={en}>
      <Store>
        <Frame>{children}</Frame>
      </Store>
    </AppProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
