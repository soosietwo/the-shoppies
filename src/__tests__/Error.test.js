import React from "react";
import { render } from "../test-utils";
import Error from "../components/Error";

describe("Error", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(
      <Error title="Test title" details="Test details" />
    );

    expect(container.querySelector(".Polaris-Banner")).toMatchSnapshot();
  });
});
