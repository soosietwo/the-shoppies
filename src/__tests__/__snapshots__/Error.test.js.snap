// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders and matches snapshot 1`] = `
<div
  aria-describedby="PolarisBanner1Content"
  aria-labelledby="PolarisBanner1Heading"
  aria-live="polite"
  class="Polaris-Banner Polaris-Banner--statusCritical Polaris-Banner--withinPage"
  role="alert"
  tabindex="0"
>
  <div
    class="Polaris-Banner__Ribbon"
  >
    <span
      class="Polaris-Icon Polaris-Icon--colorRedDark Polaris-Icon--isColored Polaris-Icon--hasBackdrop"
    >
      <svg
        aria-hidden="true"
        class="Polaris-Icon__Svg"
        focusable="false"
        viewBox="0 0 20 20"
      >
        <path
          d="M2 10c0-1.846.635-3.543 1.688-4.897l11.209 11.209A7.954 7.954 0 0 1 10 18c-4.411 0-8-3.589-8-8zm14.312 4.897L5.103 3.688A7.954 7.954 0 0 1 10 2c4.411 0 8 3.589 8 8a7.952 7.952 0 0 1-1.688 4.897zM0 10c0 5.514 4.486 10 10 10s10-4.486 10-10S15.514 0 10 0 0 4.486 0 10z"
        />
      </svg>
    </span>
  </div>
  <div
    class="Polaris-Banner__ContentWrapper"
  >
    <div
      class="Polaris-Banner__Heading"
      id="PolarisBanner1Heading"
    >
      <p
        class="Polaris-Heading"
      >
        Test title
      </p>
    </div>
    <div
      class="Polaris-Banner__Content"
      id="PolarisBanner1Content"
    >
      <p>
        Test details
      </p>
    </div>
  </div>
</div>
`;
